docs/AllRectorsOverview.md

```diff
@@ -0,0 +1,902 @@
+
+```diff
+{
+     public function someMethod()
+     {
+-        return $this->findAll();
++        return $this->repository->findAll();
+     }
+ }
+```
+
+
+## Rector\Rector\Architecture\RepositoryAsService\MoveRepositoryFromParentToConstructorRector
+
+Turns parent EntityRepository class to constructor dependency
+
+```diff
+namespace App\Repository;
+ 
++use App\Entity\Post;
+ use Doctrine\ORM\EntityRepository;
+ 
+-final class PostRepository extends EntityRepository
++final class PostRepository
+ {
++    /**
++     * @var \Doctrine\ORM\EntityRepository
++     */
++    private $repository;
++    public function __construct(\Doctrine\ORM\EntityManager $entityManager)
++    {
++        $this->repository = $entityManager->getRepository(\App\Entity\Post::class);
++    }
+ }
+```
+
+
+## Rector\Rector\Architecture\RepositoryAsService\ServiceLocatorToDIRector
+
+Turns "$this->getRepository()" in Symfony Controller to constructor injection and private property access.
+
+```diff
+class ProductController extends Controller
+ {
++    /**
++     * @var ProductRepository
++     */
++    private $productRepository;
++
++    public function __construct(ProductRepository $productRepository)
++    {
++        $this->productRepository = $productRepository;
++    }
++
+     public function someAction()
+     {
+         $entityManager = $this->getDoctrine()->getManager();
+-        $entityManager->getRepository('SomethingBundle:Product')->findSomething(...);
++        $this->productRepository->findSomething(...);
+     }
+ }
+```
+
+
+## Rector\Rector\Architecture\PHPUnit\ArrayToYieldDataProviderRector
+
+Turns method data providers in PHPUnit from arrays to yield
+
+```diff
+/**
+-                 * @return mixed[]
+                  */
+-                public function provide(): array
++                public function provide(): Iterator
+                 {
+-                    return [
+-                        ['item']
+-                    ]
++                    yield ['item'];
+                 }
+```
+
+
+## Rector\Rector\Dynamic\MethodNameReplacerRector
+
+[Dynamic] Turns method names to new ones.
+
+```diff
+$someObject = new SomeClass;
+-$someObject->oldMethod();
++$someObject->newMethod();
+ 
+-SomeClass::oldStaticMethod();
++SomeClass::newStaticMethod();
+```
+
+
+## Rector\Rector\Dynamic\PropertyToMethodRector
+
+[Dynamic] Replaces properties assign calls be defined methods.
+
+```diff
+-$result = $object->property;
+-$object->property = $value;
++$result = $object->getProperty();
++$object->setProperty($value);
+```
+
+
+## Rector\Rector\Dynamic\ClassReplacerRector
+
+[Dynamic] Replaces defined classes by new ones.
+
+```diff
+-$value = new SomeOldClass;
++$value = new SomeNewClass;
+```
+
+
+## Rector\Rector\Contrib\CodeQuality\InArrayAndArrayKeysToArrayKeyExistsRector
+
+Simplify in_array and array_keys functions combination into array_key_exists when array_keys has one argument only
```

Comment:
Can we, someway, highlight this function names?
```diff
-Simplify in_array and array_keys functions combination into array_key_exists when array_keys has one argument only
+Simplify `in_array` and `array_keys` functions combination into `array_key_exists` when `array_keys` has one argument only
```

Reply:
I thought of that too. Any ideas? 

- 1. manually 
- 2. have a list of words that should be highllighter

Reply:
Manually, like here in GitHub if we want to highlight something

Reply:
I'll update the definitions then.